hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClusterListenerSupport.java

```diff
@@ -233,7 +233,16 @@ public void run() {
                             connectToCluster();
                         } catch (Exception e) {
                             logger.warning("Could not re-connect to cluster shutting down the client", e);
-                            client.getLifecycleService().shutdown();
+                            new Thread(new Runnable() {
```

Comment:
is it possible to utilize the executor of com.hazelcast.client.impl.LifecycleServiceImpl instead of a new Thread creation here?